cpp/src/parquet/encoding.h

```diff
@@ -144,7 +145,8 @@ struct EncodingTraits<ByteArrayType> {
   /// \brief Internal helper class for decoding BYTE_ARRAY data where we can
   /// overflow the capacity of a single arrow::BinaryArray
   struct Accumulator {
-    std::unique_ptr<::arrow::BinaryBuilder> builder;
+    std::unique_ptr<::arrow::Int32Builder> offsets_builder;
+    std::unique_ptr<::arrow::BinaryBuilder> data_builder;
     std::vector<std::shared_ptr<::arrow::Array>> chunks;
   };
```

Comment:
This is taking a cue from https://github.com/apache/arrow/pull/14353#discussion_r1084323975, but it's perhaps not needed for this PR.

Reply:
Probably we don't need to this for now because `offsets_builder` is never used in this PR and the decision on how to optimize reading string/binary type is not finalized yet.

Reply:
Indeed. Removed the change.